pkg/controller/container-runtime-config/helpers.go

```diff
@@ -478,29 +489,141 @@ func validateUserContainerRuntimeConfig(cfg *mcfgv1.ContainerRuntimeConfig) erro
+			if payloadRepo != reg {
+				policyBlocked = append(policyBlocked, reg)
+			}
+			// Always add the payloadRepo to policy allowed so we can resolve for mirrors for it
+			allowed = append(allowed, payloadRepo)
+			continue
 		}
 		// Was not a match to the registry being used by the payload, so add to valid blocked registries
-		blockedRegs = append(blockedRegs, reg)
+		registriesBlocked = append(registriesBlocked, reg)
+		policyBlocked = append(policyBlocked, reg)
+	}
+	if len(blockErr) > 0 {
+		retErr = fmt.Errorf("error adding %q to blocked registries, cannot block the repository being used by the payload", blockErr)
+	}
+	return registriesBlocked, policyBlocked, allowed, retErr
+}
+
+// payloadRegHasMirrorConfig returns true if the payload registry has mirror rules configured for it
+func payloadRegHasMirrorConfig(payloadRepo reference.Named, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy, imgSpec *apicfgv1.ImageSpec) (bool, error) {
+	// Create a temp registries.conf file with all the registry inputs given
+	tomlConf := sysregistriesv2.V2RegistriesConf{}
+	if err := registries.EditRegistriesConfig(&tomlConf, imgSpec.RegistrySources.InsecureRegistries, imgSpec.RegistrySources.BlockedRegistries, icspRules); err != nil {
+		return false, err
+	}
+	var newData bytes.Buffer
+	encoder := toml.NewEncoder(&newData)
+	if err := encoder.Encode(tomlConf); err != nil {
+		return false, err
+	}
+	tmpFile, err := os.CreateTemp("", "regtemp")
+	if err != nil {
+		return false, err
+	}
+	defer os.Remove(tmpFile.Name())
+	_, err = tmpFile.Write(newData.Bytes())
+	if err != nil {
+		return false, err
+	}
+
+	var payloadMirrors []string
+	// Go through the mirror rules configured and check if a source matches the payload registry
+	// Get the possible pull sources for the payload repo using the temp registries.conf file created above
+	r, err := sysregistriesv2.FindRegistry(&types.SystemContext{SystemRegistriesConfPath: tmpFile.Name()}, payloadRepo.Name())
+	if err != nil {
+		return false, err
+	}
+	sources, err := r.PullSourcesFromReference(payloadRepo)
+	if err != nil {
+		return false, err
+	}
+
+	// Go through the pull sources and grab their mirrors
+	// Also check if any of them are in the blocked list
+	for _, s := range sources {
+		for _, blockReg := range imgSpec.RegistrySources.BlockedRegistries {
+			if s.Reference.Name() == blockReg {
+				payloadMirrors = getMirrors(s.Reference.Name(), icspRules)
+				// If a source is blocked and there are no mirrors for it, return false
+				if len(payloadMirrors) == 0 {
+					return false, nil
+				}
+			}
+		}
+		// Grab the mirrors for each pull source
+		payloadMirrors = append(payloadMirrors, getMirrors(s.Reference.Name(), icspRules)...)
+	}
+
+	var isUsable bool
+	// Go through the list of mirrors and check that at least one of the payload mirrors is not blocked
+	for _, mirror := range payloadMirrors {
```

Comment:
… this could be _even more_ c/image-like as
```go
reg2 := sysregistriesv2.FindRegistry(ourConfig, mirror);
isUsable :=  reg2 == nil || !reg2.Blocked
```
but AFAICS the two should be equivalent.